test/fixtures/es-module-loaders/loader-with-custom-condition.mjs

```diff
@@ -0,0 +1,6 @@
+export async function resolve(specifier, context, defaultResolve) {
+  const customContext = Object.assign({}, context, {
+    conditions: ['custom-condition'].concat(context.conditions),
+  });
+  return defaultResolve(specifier, customContext);
```

Comment:
`return defaultResolve(specifier, { ...context, conditions: ['custom-condition', ...context.conditions] })` might be simpler.

Reply:
Updated. This also showed that `conditions` wasn't provided to the loader, updated that as well. The loader now gets a frozen default conditions array.